### DISCUSSION TITLE
Hosted chatbot-ui security

### DISCUSSION THREAD

**Discussion by 11hans** [mckaywrigley/chatbot-ui#1599] [Help]
Hi, I apologize if this seems like a simple question, but I've been unable to find an answer. I'm wondering where the service_role KEY is stored when I host chatbot-ui on Vercel from my Github repository. If I understand correctly, I entered it into Vercel, so does that mean that line 54 in supabase/migration/setup.sql should be empty? Or am I missing something? Thank you for your help; I'm just starting out.

**Comment by spammenotinoz** [comment]
On vercel you don't want to use .env.local for security reasons, so you use environment variables (found within settings).
You also need to update line 54 in the migrations file and set the project path url in the migrations file.  (PS: You can run the migrations file afterwards if you got it wrong), the migrations file impacts file uploads and user deletions if it is incorrect.
